haystack/preview/dataclasses/answer.py

```diff
@@ -0,0 +1,8 @@
+from typing import Optional
+from dataclasses import dataclass
+
+
+@dataclass
+class Answer:
+    answer: Optional[str]
+    probability: float
```

Comment:
Should this contain question, document, start and end?

Reply:
That is actually a very good question. I'd say all of them are useful, but then we might need an extra field that specifies whether it's generative, so it does not contain start and end, or extractive, so it does.

Maybe let's do the following:

```
@dataclass
class Answer:
    answer: Optional[str]
    probability: float
    question: str
    documents: List[Document]   # might be more than one if it's a generative answer
    positions = List[Span]    # we might need to also include something different for tables
```

Reply:
However I think the `Answer` object is worth its own small PR

Reply:
Superseded by https://github.com/deepset-ai/haystack/pull/5563